lib/iris/__init__.py

```diff
@@ -303,7 +312,118 @@ def _load_collection(uris, constraints=None, callback=None):
     return result
 
 
-def load(uris, constraints=None, callback=None):
+class LoadPolicy(threading.local):
+    """Object defining a general loading strategy."""
+
+    _allkeys = (
+        "support_multiple_references",
+        "multiref_triggers_concatenate",
+        "use_concatenate",
+        "use_merge",
+        "cat_before_merge",
+        "repeat_until_done",
```

Comment:
I like the idea of giving users more controls to play with, but having 6 booleans is very confusing, so I'm keen to look for ways it could be simpler to understand.

1. It doesn't look like `support_multiple_references` is necessary
2. `use_concatenate`, `use_merge` and `cat_before_merge` could become some sort of controllable sequence e.g. `{"c", "m"}` / `{"m"}` / `{"m", "c"}`.

Reply:
OK, as discussed offline, I think we can simplify all this with benefits.